### DISCUSSION TITLE
Welcome to gcp-bigquery-client Discussions!

### DISCUSSION THREAD

**Discussion by lquerel** [lquerel/gcp-bigquery-client#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Plommonsorbet** [comment]
Hey!

Thanks for creating this library, I'm really happy to be able to use rust for working with bigquery data!

I have a question about using the query parameter option:

Looking through the docs I've found that there is the option `QueryParameter` however it requires you to declare a `QueryParameterType` which I'm not quite clear on how to do.

Looking at the code snippet for the `QueryParameterType`, I'm understanding this as a recursive type since there is no Option around `array_type`, I know this is normally fine because of the Box but I don't understand how to opt out of this without an option or some other kind of enum or break point. 
```rust
pub struct QueryParameterType {
    pub array_type: Box<QueryParameterType>,
    pub struct_types: Option<Vec<QueryParameterTypeStructTypes>>,
    pub type: String,
}
```

Maybe there is some trick I don't know with Box to not make it infinitely recursive? Would love to just get a short code snippet of how to define a simple parameter type like a string or int if that would be possible?

**Comment by crbl1122** [comment]
Hi! I am looking at the example provided for gcp_bigquery_client but I have an authentication issue. I am running the code from a GCP Compute Engine and I am using a service account to access resources (like BigQuery), not a credentials file. The example I see is only for credentials file usage. Could someone provide an example about how I can authenticate with this client into BigQuery using a service account: username@project_id.iam.gserviceaccount.com? Thank you.
